GoogleSignIn/Sources/GIDGoogleUser.m

```diff
@@ -184,11 +184,11 @@ - (OIDAuthState *) authState{
 }
 
 - (void)addScopes:(NSArray<NSString *> *)scopes
-#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+#if TARGET_OS_IOS || TARGET_OS_MACCATALYST
     presentingViewController:(UIViewController *)presentingViewController
-#elif TARGET_OS_OSX || TARGET_OS_MACCATALYST
+#elif TARGET_OS_OSX
             presentingWindow:(NSWindow *)presentingWindow
-#endif // TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+#endif // TARGET_OS_IOS || TARGET_OS_MACCATALYST
```

Comment:
@Alex-4-Git note that you missed an instance of the same platform check mistake below and the build action is now failing for that reason:

https://github.com/google/GoogleSignIn-iOS/actions/runs/3572147376/jobs/6004706482